### ISSUE TITLE
⌨ Tables lack a label

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#9379]
## ⌨ Accessibility Issue

It looks like our tables have the `aria-labelledby` applied to the wrong `div` 😢 

### WCAG 2.1 Criteria

- https://dequeuniversity.com/rules/axe/4.8/landmark-unique?application=AxeChrome
- https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/region.html

## 🦋 Expected Behaviour

Table regions have a name in the accessibility tree and are unique.

## 🕵️ Details

Looks like we are applying the `aria-labelledby` to the main wrapping `div` but the `role="region"` is applied to a child `div`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/a66841ef459952763f3b409238d2b2329f42cd00/apps/web/src/components/Table/ResponsiveTable/ResponsiveTable.tsx#L330-L331

https://github.com/GCTC-NTGC/gc-digital-talent/blob/a66841ef459952763f3b409238d2b2329f42cd00/apps/web/src/components/Table/ResponsiveTable/Table.tsx#L20-L21

## 📋 Steps to Reproduce

1. Navigate to the skill library
2. Open the accessibility tree
3. Inspect the region wrapping the tables
4. Observe that there is no name

## 📸 Screenshot

![Screenshot 2024-02-15 131732](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/80a9fd5b-665c-4c3e-8bbd-f41005a624aa)

## 🙋‍♀️ Proposed Solution

I beleive we should be able to just move the `aria-labelledby` and `role="region"` so they are on the same element.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Table regions have programmatically discernible name
- [ ] Table region names are unique
